topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -172,7 +186,26 @@ public void start(Promise<Void> start) {
                 LOGGER.debug("Using ZooKeeper {}", zk);
 
                 String topicsPath = config.get(Config.TOPICS_PATH);
-                ZkTopicStore topicStore = new ZkTopicStore(zk, topicsPath);
+                TopicStore topicStore;
+                if (config.get(Config.USE_ZOOKEEPER_TOPIC_STORE)) {
+                    topicStore = new ZkTopicStore(zk, topicsPath);
+                } else {
+                    // TODO -- better async handling?
```

Comment:
what about this comment?

Reply:
I thought TopicOperator could somehow use async result of TopicStore creation, but I guess that's not the case, hence forcing the result with ConcurrentUtil::result.
If anyone has some better async ideas ...